docs/add-annotation-type.md

```diff
@@ -0,0 +1,27 @@
+# Adding support for new Annotation types
+See [Image Point Annotations](https://github.com/box/box-annotations/tree/master/src/image) for a basic implementation of an annotation type.
+
+### The following files/abstract methods need to be implemented:
+* controllers/{**TYPE**}Controller.js (extends [AnnotationModeController](https://github.com/box/box-annotations/blob/master/src/controllers/AnnotationModeController.js)): This class is the communicator between the Annotator and the Annotation threads. Each controller maintains an object of pages which contains a [red/black tree](https://github.com/mourner/rbush)
+    * Abstract methods that should be implemented by subclasses:
+        * **handleThreadEvents()** - Handles annotation [thread events](thread.md#events) and emits them to the viewer
+        * **enter()** - Enables the specified annotation mode
+        * **exit()** - Disables the specified annotation mode
+
+* {**VIEWER**}/{**TYPE**}Dialog.js (extends [AnnotationDialog](https://github.com/box/box-annotations/blob/master/src/AnnotationDialog.js)): This class maintains the dialog UI corresponding with an annotation thread, displaying the metadata of annotations and buttons to create/update/delete
```

Comment:
Maybe have a note briefly explaining what a viewer is?

Reply:
Added a link to preview documentation